pkg/processor/trigger/http/trigger.go

```diff
@@ -211,7 +213,75 @@ func (h *http) AllocateWorkerAndSubmitEvent(ctx *fasthttp.RequestCtx,
 	return response, false, nil, processError
 }
 
-func (h *http) requestHandler(ctx *fasthttp.RequestCtx) {
+func (h *http) handleRequest() fasthttp.RequestHandler {
+	return func(ctx *fasthttp.RequestCtx) {
+		if h.configuration.CORS.Enabled &&
+			common.ByteSlice2String(ctx.Method()) == h.configuration.CORS.PreflightRequestMethod {
+
+			// when CORS is enabled, processor HTTP server is responding to "PreflightRequestMethod" (e.g.: OPTIONS)
+			// That means => function will not be able to answer on the method configured by PreflightRequestMethod
+			h.handlePreflightRequest(ctx)
+		} else {
+			h.handleActualRequest(ctx)
+		}
+	}
+}
+
+func (h *http) handlePreflightRequest(ctx *fasthttp.RequestCtx) {
+
+	// default to bad preflight request unless all specifications are valid
+	ctx.SetStatusCode(fasthttp.StatusBadRequest)
+
+	origin := common.ByteSlice2String(ctx.Request.Header.Peek("Origin"))
+
+	// ensure origin is given, otherwise the request is outside the scope of CORS specifications
+	if !h.configuration.CORS.IsSetAndMatchOrigin(origin) {
+		return
+	}
+
+	method := common.ByteSlice2String(ctx.Request.Header.Peek("Access-Control-Request-Method"))
+
+	// ensure method is given, otherwise the request is outside the scope of CORS specifications
+	if !h.configuration.CORS.IsSetAndMatchMethod(method) {
+		return
+	}
+
+	headers := common.ByteSlice2String(ctx.Request.Header.Peek("Access-Control-Request-Headers"))
+	if headers != "" {
+
+		// ensure request headers allowed (it can also be empty)
+		if !h.configuration.CORS.AreMatchHeaders(strings.Split(headers, ", ")) {
+			return
+		}
+	}
+
+	// indicate whether resource can be shared
+	ctx.Response.Header.Set("Access-Control-Allow-Origin",
```

Comment:
1. no need for newline here
2. Note that under wildcard, you will return the *specific* host from the origin rather than "*" which is configured. I think this is good behavior, but theoretically you are also returning a narrower response than configured. Are we OK with this? Maybe this should be configurable (i.e. wildcardResponseMode: origin / static)

Reply:
1. 👍 
2. it is as expected by w3 CORS, also, returning "*" is not valid when returning "true" for "Allow-Credentials", so adding that will complex other stuff (_The string "*" cannot be used for a resource that supports credentials_ from their docs)